enterprise/app/repo/repo.tsx

```diff
@@ -149,20 +155,44 @@ export default class RepoComponent extends React.Component<RepoComponentProps, R
     );
   }
 
+  getSecrets() {
+    let s = this.props.search.get("secret") || this.props.search.get("secrets") || "";
+    return s.split(",");
+  }
+
   async handleCreateClicked() {
     this.setState({ isCreating: true });
     try {
       await this.linkInstallation();
       let repoResponse = await this.createRepo();
       this.setState({ repoResponse: repoResponse });
-      let workflowResponse = await this.runWorkflow(repoResponse.repoUrl);
-      this.setState({ isCreating: false, workflowResponse: workflowResponse });
+      if (!this.getSecrets()) {
+        this.handleDeployClicked(repoResponse);
+      }
     } catch (e) {
       error_service.handleError(e);
       this.setState({ isCreating: false });
     }
   }
 
+  async saveSecrets() {
+    await this.getSecrets().map((s) => encryptAndUpdate(s, this.state.secrets.get(s) || ""));
```

Comment:
I think this will return immediately since this `map()` returns `Array` not `Promise`.

```suggestion
    await Promise.all(this.getSecrets().map((s) => encryptAndUpdate(s, this.state.secrets.get(s) || "")));
```